apps/etterlatte-saksbehandling-ui/client/src/components/klage/vurdering/KlageVurderingFelles.tsx

```diff
@@ -0,0 +1,127 @@
+import { Klage, TEKSTER_AARSAK_OMGJOERING, TEKSTER_LOVHJEMLER } from '~shared/types/Klage'
+import React, { useRef } from 'react'
+import { useApiCall } from '~shared/hooks/useApiCall'
+import { hentBrev } from '~shared/api/brev'
+import { isFailure, isInitial, mapApiResult } from '~shared/api/apiUtils'
+import { Alert, BodyShort, Button, Heading, Modal } from '@navikt/ds-react'
+import Spinner from '~shared/Spinner'
+import { ApiErrorAlert } from '~ErrorBoundary'
+import ForhaandsvisningBrev from '~components/behandling/brev/ForhaandsvisningBrev'
+import styled from 'styled-components'
+import { JaNei } from '~shared/types/ISvar'
+
+export function VisInnstilling(props: { klage: Klage; kanRedigere: boolean }) {
+  const { klage, kanRedigere } = props
+  const innstilling =
+    klage.utfall?.utfall === 'DELVIS_OMGJOERING' || klage.utfall?.utfall === 'STADFESTE_VEDTAK'
+      ? klage.utfall.innstilling
+      : null
+  const ref = useRef<HTMLDialogElement>(null)
+  const [brev, hentBrevet] = useApiCall(hentBrev)
+
+  function visModal() {
+    if (!innstilling) return
+
+    if (isInitial(brev) || isFailure(brev)) {
+      hentBrevet({ sakId: klage.sak.id, brevId: innstilling.brev.brevId })
+    }
+    ref.current?.showModal()
+  }
+
+  if (!innstilling) return null
+
+  return (
+    <Maksbredde>
+      <Heading size="small" level="3">
+        Innstilling til KA
+      </Heading>
+      <BodyShort spacing>
+        Vedtak opprettholdes med følgende hovedhjemmel: <strong>{TEKSTER_LOVHJEMLER[innstilling.lovhjemmel]}.</strong>
+      </BodyShort>
+      <BodyShort spacing>
+        <Button size="small" variant="primary" onClick={visModal}>
+          Se innstillingsbrevet
+        </Button>
+      </BodyShort>
+
+      {kanRedigere && (
+        <Alert variant="info">
+          Når klagen ferdigstilles vil innstillingsbrevet bli sendt til mottaker, og klagen blir videresendt for
+          behandling av KA.
+        </Alert>
+      )}
+
+      <Modal width="medium" ref={ref} header={{ heading: 'Innstillingsbrev' }}>
+        <Modal.Body>
+          {mapApiResult(
+            brev,
+            <Spinner visible label="Laster brevet" />,
+            () => (
+              <ApiErrorAlert>Kunne ikke hente brevet, prøv å laste siden på nytt.</ApiErrorAlert>
+            ),
+            (hentetBrev) => (
+              <ForhaandsvisningBrev brev={hentetBrev} />
+            )
+          )}
+        </Modal.Body>
+      </Modal>
+    </Maksbredde>
+  )
+}
+
+export function VisOmgjoering(props: { klage: Klage; kanRedigere: boolean }) {
```

Comment:
Hvorfor endre denne fra å ta inn omgjøring til å ta inn klage? Nå må det være mer logikk for å hente ut omgjøringen, og komponenten skjuler seg selv hvis du sender inn noe feil. Da tenker jeg kompileringsfeil hvis du sender noe som ikke er omgjøring + "automatisk" mapping er bedre

Reply:
Ok. Var usikker, men det var litt for å slippe å gjenta dette: {klage.utfall?.utfall === 'DELVIS_OMGJOERING' || klage.utfall?.utfall === 'STADFESTE_VEDTAK' ? ... }
Kanskje jeg heller kan lage en metode "harInnstilling" eller noe? Er det en "regel" at en komponent ikke skal returnere null?

Reply:
Det er ikke nødvendigvis en regel at komponenten ikke skal returnere null, men det er ofte lettere å forstå hva som vises og ikke hvis komponenter i hovedsak unngår det. Hvis man har en komponent A som kun skal vises hvis n er partall kan man forstå dette med en gang hvis den brukes slik:
```
{n % 2 === 0 && <A n={n} />}
```
mens en 
```
<A n={n} />
```
betyr at man må gå inn i komponenten A for å skjønne om den skal vises eller ikke. 

Ikke en absolutt regel altså, men det gjør ofte koden lettere å lese hvis komponenter ikke skjuler seg selv (med en haug med unntak da)